packages/atclient/src/atkey.c

```diff
@@ -73,19 +73,22 @@ size_t atclient_atkey_strlen(const atclient_atkey *atkey) {
     atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "atkey is NULL\n");
     return 0;
   }
```

Comment:
Separate if-clauses that check if 'key' and 'shared_by' are initialized. Also, accept if reserved keys do not have shared_by initialized